addons/faster-project-loading/userscript.js

```diff
@@ -1,13 +1,12 @@
 // This is a fix for https://github.com/scratchfoundation/scratch-gui/issues/8805
 
+import { BACKPACK_URL, SPRITE_FILE_EXTENSION, isBadRequest } from "./module.js";
+
 export default async function ({ addon }) {
-  const BACKPACK_URL = "https://backpack.scratch.mit.edu/";
-  // Inserting sprites from the backpack requests a ZIP archive from backpack.scratch.mit.edu, so we want to allow those
-  const SPRITE_FILE_EXTENSION = ".zip";
   // Dropping a code item from the backpack into a specific sprite within the sprite-pane (NOT into the code area)
   // requests a JSON file (https://backpack.scratch.mit.edu/{hash}.json) which we shouldn't block
   const CODE_FILE_EXTENSION = ".json";
```

Comment:
Why is `CODE_FILE_EXTENSION` outside the module?

Reply:
Oh, was `SPRITE_FILE_EXTENSION` supposed to be outside the module too?

Reply:
Sorry, I'm still trying to figure out what's happening here.

Reply:
Sorry just changed some things, it's confusing but the bad backpack requests are made twice, once using a webworker and the other using XHR. When trapping all of XHR, we have to deal with all the requests Scratch is making with XHR and thus there's a lot more edge cases. The web worker only deals with assets, thus the ZIP and JSON edge cases are not ever sent to the web worker, so they don't need to be check for.

Reply:
Oh yeah, because the file extension conditions don't need to be accounted for during project loading.